SDK_versioned_docs/version-3.0.0/guides/auto-router/swap-and-add.md

```diff
@@ -0,0 +1,147 @@
+	position: Position,
+	swapAndAddConfig: SwapAndAddConfig,
+	swapAndAddOptions?: swapAndAddOptions,
+	partialRoutingConfig?: Partial<AlphaRouterConfig> = {}
+}
+```
+
+#### Parameters
+
+`token0Balance` [required]
+
+- The initial starting balance of token0 of the pool for which to add liquidity
+
+`token1Balance` [required]
+
+- The initial starting balance of token1 of the pool for which to add liquidity
+
+`position` [required]
+
+- A position object that contains the details of the position for which to add liquidity. The position liquidity can be set to 1, since liquidity is still unknown and will be set inside the call to `routeToRatio`
+
+`swapAndAddConfig` [required]
+
+- Configurations for the routeToRatio algorithm. `ratioErrorTolerance` determines the margin of error the resulting ratio can have from the optimal ratio. `maxIterations` determines the maximum times the algorithm will iterate to find a ratio within error tolerance. If max iterations is exceeded, an error is returned.
+
+`swapAndAddOptions` [optional]
+
+- If included, routeToRatio will return the calldata for executing the atomic swap-and-add. These options contain `swapConfig` and `addLiquidityOptions`. `swapConfig` configures to set a recipient of leftover dust from swap, slippageTolerance, deadline, and inputTokenPermit. `addLiquidityOptions` contains slippageTolerance and deadline for the swap. `addLiquidityOptions` must contain a `tokenId` to add to an existing position, or `recipient` to mint a new one. It also includes a slippage tolerance and deadline for adding liquidity.
+
+`routingConfig` [optional]
+
+- Optional config for tuning the performance of the routing algorithm.
+
+```typescript
+const USDC = new Token(
+  ChainId.MAINNET,
+  '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+  6,
+  'USDC',
+  'USD//C'
+);
+
+const WETH = new Token(
+  1,
+  '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+  18,
+  'WETH',
+  'Wrapped Ether'
+);
+
+const token0Balance = CurrencyAmount.fromRawAmount(USDC, '5000000000')
+const token1Balance = CurrencyAmount.fromRawAmount(WETH, '0')
+const pool = new Pool(
+  USDC,
+  WETH,
+  3000,
+  '1283723400872544054280619964098219',
+  '8390320113764730804' ,
+  '193868'
+);
+
+
+const route = await router.route({
+	token0Balance,
+	token1Balance,
+	position: new Position({
+		pool,
+		tickLower: -60,
+		tickUpper: 60,
+		liquidity: 1, // arbitrary and unused
+	}),
+	swapAndAddConfig: {
+		swapConfig: {
+			recipient: myAddress,
+			slippage: new Percent(5, 100),
+			deadline: 100
+		},
+		addLiquidityOptions: {
+			tokenId: 10,
+			slippage: new Percent(5, 100),
+			deadline: 100
+		}
+	}
+);
+```
+
+## Submitting a Transaction
+
+The object returned from calling `routeToRatio` is a `SwapToRatioResponse`. If a route was found successfully, `SwapToRatioResponse` will have to fields: `status` and `result`. `status` will be set to success and `result` contains the `SwapToRatioRoute` object.
+
+```typescript
+export type SwapRoute = {
```

Comment:
what is this? The doc should explain what it is, why we use it, and where it is used.  I don't see it used in the following code example?

Maybe it's the parent class of `SwapToRatioRoute`? If so let's make that clear with something like "The SwapToRatioRoute object is of type `SwapRoute` with the following parameters.